File: api/extract.js
if (req.method !== "POST") {
    res.status(405).json({ error: "Method Not Allowed" });
    return;
  }

  // Collect the PDF as a buffer
  const chunks = [];
  for await (const chunk of req) {
    chunks.push(chunk);
  }
  const buffer = Buffer.concat(chunks);

  try {
    const data = await pdfParse(buffer);
    res.status(200).json({
      numPages: data.numpages,
      text: data.text.slice(0, 1000), // preview
    });
  } catch (err) {
    res.status(500).json({ error: err.toString() });
  }

File: api/extract.py
import pdfplumber
import io
import json
import base64

def handler(request):
    # Only allow POST
    if request.method != "POST":
        return {
            "statusCode": 405,
            "body": json.dumps({"error": "Method Not Allowed"}),
            "headers": {"Content-Type": "application/json"}
        }

    # Parse form-data; Vercel exposes uploaded files in request.files
    try:
        form = request.files
        if not form or "file" not in form:
            return {
                "statusCode": 400,
                "body": json.dumps({"error": "No file uploaded"}),
                "headers": {"Content-Type": "application/json"}
            }
        pdf_file = form["file"]
        pdf_bytes = pdf_file.read()
        # Simple processing: count pages
        with pdfplumber.open(io.BytesIO(pdf_bytes)) as pdf:
            num_pages = len(pdf.pages)
        return {
            "statusCode": 200,
            "body": json.dumps({"num_pages": num_pages}),
            "headers": {"Content-Type": "application/json"}
        }
    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"error": str(e)}),
            "headers": {"Content-Type": "application/json"}
        }